metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java

```diff
@@ -394,6 +399,52 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(MetisUserView me
     // Done.
     return new RecordsResponse(records);
   }
+  /**
+   * Get a record from the external resource based on o searchId, workflow execution and {@link PluginType}.
+   *
+   * @param metisUserView the user wishing to perform this operation
+   * @param workflowExecutionId the execution identifier of the workflow
+   * @param pluginType the {@link ExecutablePluginType} that is to be located inside the workflow
+   * @param searchId the searchId of the record we wish to obtain
+   * @return the record from the external resource
+   * @throws GenericMetisException can be one of:
+   * <ul>
+   * <li>{@link MCSException} if an error occurred while retrieving the records from the external
+   * resource</li>
+   * <li>{@link eu.europeana.metis.exception.UserUnauthorizedException} if the user is not
+   * authorized to perform this task</li>
+   * <li>{@link eu.europeana.metis.core.exceptions.NoWorkflowExecutionFoundException} if no workflow
+   * execution exists for the provided identifier</li>
+   * </ul>
+   */
+  public Record lookupIdFromUISClient(MetisUserView metisUserView, String workflowExecutionId , ExecutablePluginType pluginType, String searchId )
+      throws GenericMetisException {
+
+    // Get the right workflow execution and plugin type.
+    final Pair<WorkflowExecution, AbstractExecutablePlugin> executionAndPlugin = getExecutionAndPlugin(
+        metisUserView, workflowExecutionId, pluginType);
+    if (executionAndPlugin == null) {
+      throw new NoWorkflowExecutionFoundException(String
+          .format("No executable plugin of type %s found for workflowExecution with id: %s",
+              pluginType.name(), workflowExecutionId));
+    }
+    // Authorize
+//    authorizer.authorizeReadAllDatasets(metisUserView);
+
+    // lookup for eCloudId.
+    final CloudId cloudId;
+    try {
+      cloudId = uisClient.getCloudId(searchId,searchId);
```

Comment:
I think there may be the case that the searchId is ALREADY a cloud ID. I think that in this case `uisClient.getCloudId()` would return null, but in that case, you should set `cloudId = searchId` and try with that.
Once deployed, we should test this.

Reply:
I don't think this makes much sense. But perhaps I'm missing something.
`    CloudId cloudId = null;
    try {
      cloudId = uisClient.getCloudId(searchId,searchId);
      if (cloudId != null) {
        return Optional.ofNullable(getRecord(executionAndPlugin.getRight(), cloudId.getId())).orElse(null);
      }else {
        cloudId.setId(searchId);
        return Optional.ofNullable(getRecord(executionAndPlugin.getRight(), cloudId.getId())).orElse(null);
      }`

Reply:
Fixed.